src/pages/storage/HostPathDeviceModal.tsx

```diff
@@ -0,0 +1,98 @@
+import { FC, ReactNode, useRef, useState } from "react";
+import { Button, Input, Modal } from "@canonical/react-components";
+import { InstanceAndProfileFormikProps } from "components/forms/instanceAndProfileFormValues";
+import { LxdDiskDevice } from "types/device";
+
+interface Props {
+  formik: InstanceAndProfileFormikProps;
+  onFinish: (device: LxdDiskDevice) => void;
+  onCancel: () => void;
+  title?: ReactNode;
+}
+
+const HostPathDeviceModal: FC<Props> = ({
+  formik,
+  onFinish,
+  onCancel,
+  title,
+}) => {
+  const [source, setSource] = useState("");
+  const [path, setPath] = useState<string>();
+  const touchedRef = useRef({
+    source: false,
+    path: false,
+  });
```

Comment:
Why use a ref here and not a normal state?

Reply:
They could be, used ref here since it does not cause unnecessary renders